source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems,
+       Nix community has chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is Open Source. 
+          It's only possible to cross-compile between ``aarch64-darwin`` and ``x86_64-darwin``.
+
+
+What's a target platform?
+-------------------------
+
+There's actually a third platform named target.
+
+It matters in cases where you'd like to distribute a compiler binary, 
+as you'd then like to build a compiler on the build platform, compile code on the
+target plaform and run the final executable on the host platform.
+
+Since that's rarely needed, we'll treat target platform the same as the build.
+
+
+Determining the host platform
+-----------------------------
+
+The build platform is determined automatically by Nix
+as it can just guess it during the configure phase.
+
+The host platform is best determined by running on the host platform:
+
+.. code:: shell-session 
+
+  $ bash $(nix-build '<nixpkgs>' -A gnu-config)/config.guess
+  aarch64-unknown-linux-gnu
+
+In case that's not possible (when the host platform is not easily accessible
+for development), it has to be constructed manually via the following template:
+
+.. code::
+
+  <cpu>-<vendor>-<os>-<abi>
+
+Note that ``<vendor>`` is often ``unknown`` and ``<abi>`` is optional. 
+There's also no unique identifier for a platform, for example ``unknown`` and 
+``pc`` are interchangeable (hence it's called config.guess).
+
+Some other common examples of platforms:
+
+- aarch64-apple-darwin14
+- aarch64-pc-linux-gnu
+- x86_64-w64-mingw32
+- aarch64-apple-ios
+
```

Comment:
This section feels superfluous if you never mention the `crossSystem` argument which can actually be used to pass a `config` such as we have gotten in this section to use as the host platform for nixpkgs.

Mentioning `crossSystem` is important because it is the most basic case, i. e. `pkgsCross` is just a convenient shortcut for (re-)importing nixpkgs with `crossSystem` set.

Reply:
How often do we imagine the user provides their own crossSystem? Currently I want to teach that the single truth is the platform identifier and how it one can go between nixpkgs attributes and the identifier.

In case the user needs a new crossSystem, there's a note how to add one to nixpkgs.

Can you think of a use case where that's not enough?

Reply:
Well the immediate use case would of course be (as it has come up here) that you can pass a triple to `crossSystem` and it will figure things out (or try to).

Reply:
Good point :laughing: